src/util/file_util.cpp

```diff
@@ -36,29 +39,19 @@ Date: January 2012
 #define chdir _chdir
 #define popen _popen
 #define pclose _pclose
-#else
-#include <cstring>
 #endif
 
 /// \return current working directory
 std::string get_current_working_directory()
 {
-  unsigned bsize=50;
-
-  char *buf=reinterpret_cast<char*>(malloc(sizeof(char)*bsize));
-  if(!buf)
-    abort();
-
   errno=0;
+  char *wd=realpath(".", nullptr);
+  INVARIANT(
+    wd!=nullptr && errno==0,
```

Comment:
CHECK_RETURN?

Reply:
That would not permit an error message, which is descriptive in this case.